### ISSUE TITLE
Internal error for non-text responses

### ISSUE THREAD

**Issue by kevinoid** [swagger-api/swagger-js#979]
Responses with a non-text `Content-Type` cause `swagger-client` to fail with an uninformative error.  For example:

```js
'use strict';

const assert = require('assert');
const http = require('http');
const SwaggerClient = require('swagger-client');

const TEST_HOST = 'localhost';
const TEST_ID = 123;
const TEST_PORT = 8888;

http
  .createServer((req, res) => {
    assert.strictEqual(req.url, '/binary-data');
    res.writeHead(200, {'Content-Type': 'application/octet-stream'});
    res.end(new Buffer([1, 2, 3, 4, 5]));
  })
  .listen(TEST_PORT, TEST_HOST, () => {
    new SwaggerClient({
      usePromise: true,
      spec: {
        swagger: '2.0',
        info: {
          title: 'Path Parameter API',
          version: '1.0.0'
        },
        host: `${TEST_HOST}:${TEST_PORT}`,
        schemes: [
          'http'
        ],
        paths: {
          '/binary-data': {
            get: {
              operationId: 'getBinaryData',
              responses: {
                200: {
                  description: 'Some binary data',
                  schema: {
                    type: 'file'
                  }
                }
              }
            }
          }
        }
      }
    })
      .then(client => client.apis.default.getBinaryData())
      .then(
        () => {
          console.log('It works.');
          process.exit();
        },
        err => {
          console.error(err);
          process.exit(1);
        }
      );
  });
```
Produces the following error:

    TypeError: oriRes[(intermediate value)(intermediate value)(intermediate value)] is not a function
        at Object.serializeRes (dist/index.js:443:43)
        at dist/index.js:414:17
        at process._tickCallback (internal/process/next_tick.js:103:7)

The cause appears to be that [`node-fetch`](https://github.com/bitinn/node-fetch) (used by [`isomorphic-fetch`](https://github.com/matthew-andrews/isomorphic-fetch)) only supports the `.blob` response property in version 2 (which is currently in alpha) causing the failure [when `useText` is `false` and `.blob` is called in serializeRes](https://github.com/swagger-api/swagger-js/blob/v3.0.3/src/http.js#L68).

I don't see an easy solution.  Using `body.buffer()` when `body.blob` is not present would work, but would cause API divergence between Node and browser versions.  Depending on `node-fetch` 2.0 alpha would work, but adds instability risk and would require reimplementing the tricks of `isomorphic-fetch` to get isomorphism.  Including the `Blob` implementation from `node-fetch` 2.0 alpha for use with `node-fetch` 1 until 2.0 is released would work but would introduce more code and maintenance burden.  Thoughts?

Thanks,
Kevin

**Comment by buunguyen** [comment]
@kevinoid thanks for the detailed report. This makes a lot of sense.

I think we should go with `body.buffer()` for now then make change when node-fetch 2.0 is released. Would you mind submitting a PR+test for this?

**Comment by kevinoid** [comment]
Sure.  Proposed fix in PR #985.
